### ISSUE TITLE
Something odd is happening on my cluster's reverse proxies and I think Authelia is causing it

### ISSUE THREAD

**Issue by Slyke** [authelia/authelia#3299]
### Description

Every month, at the start of the month my cluster reboots each node. It does this over the course of a day to ensure stability. Cluster has been running fine for months without being modified, but as of the start of this month, there are lots of random things failing. At first I thought it was nginx, but not so sure now. Each time I try to troubleshoot issues, I can get different errors, but as of right now I think it's being caused by Authelia.

### Expected Behaviour

When running the command to the nginx reverse proxy that Authelia is configured for:
```
curl -k -H "Host: XXXX.XXX" http://10.7.9.20/motioneye/
```

I get the following errors showing up on nginx:
```
2022/05/04 09:07:29 [error] 31#31: *199 connect() failed (110: Operation timed out) while connecting to upstream, client: 10.244.0.0, server: _, request: "GET /motioneye/ HTTP/1.1", subrequest: "/authelia-auth", upstream: "http://10.7.9.10:9091/api/verify", host: "XXXX.XXX"
2022/05/04 09:07:29 [error] 31#31: *199 auth request unexpected status: 502, client: 10.244.0.0, server: _, request: "GET /motioneye/ HTTP/1.1", host: "XXXX.XXX"
10.244.0.0 - - [04/May/2022:09:07:29 +0000] "GET /motioneye/ HTTP/1.1" 500 177 "-" "curl/7.74.0" "-"
```

If I have Authelia configured to use LDAP I get this error:
```
ala/fasthttp@v1.35.0/workerpool.go:224                       (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.35.0/workerpool.go:196                       (*workerPool).getCh.func1\nruntime/asm_amd64.s:1571                                                    goexit"
time="2022-05-04T08:19:08Z" level=error msg="Error caught when verifying user authorization: LDAP Result Code 200 \"Network Error\": dial tcp 10.7.9.10:389: i/o timeout" method=GET path=/api/verify remote_ip=10.244.2.105 stack="github.com/authelia/authelia/v4/internal/handlers/handler_verify.go:478     VerifyGET.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/authelia_context.go:53 AutheliaMiddleware.func1.1\ngithub.com/fasthttp/router@v1.4.8/router.go:414                             (*Router).Handler\ngithub.com/valyala/fasthttp@v1.35.0/http.go:153                             (*Response).StatusCode\ngithub.com/authelia/authelia/v4/internal/middlewares/strip_path.go:21       StripPathMiddleware.func1\ngithub.com/valyala/fasthttp@v1.35.0/server.go:2338                          (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.35.0/workerpool.go:224                       (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.35.0/workerpool.go:196                       (*workerPool).getCh.func1\nruntime/asm_amd64.s:1571
```

If I switch to use yaml file, I get nothing in the Authelia logs when running the same curl command (but still get the same nginx log error):
```
time="2022-05-04T09:04:11Z" level=info msg="Authelia v4.35.1 is starting"
time="2022-05-04T09:04:16Z" level=warning msg="Could not read from the NTP server socket to validate the system time is properly synchronized: read udp 10.244.2.111:51132->10.7.9.10:123: i/o timeout"
```

### Reproduction Steps

_N/A_

### Additional Information
I noticed that when navigating to my XXXX.XXX/motioneye in the browser, I would get a 308 redirect to XXXX.XXX/, but I suspect this is nginx.

Auth.config:
```
auth_request /authelia-auth;

auth_request_set $target_url $scheme://$http_host$request_uri;

auth_request_set $user $upstream_http_remote_user;
auth_request_set $groups $upstream_http_remote_groups;
auth_request_set $name $upstream_http_remote_name;
auth_request_set $email $upstream_http_remote_email;
proxy_set_header Remote-User $user;
proxy_set_header Remote-Groups $groups;
proxy_set_header Remote-Name $name;
proxy_set_header Remote-Email $email;

error_page 401 =302 https://$host/authelia/?rd=$target_url;
```

Proxy.config:
```
client_body_buffer_size 128k;

#Timeout if the real server is dead
proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;

# Security
add_header X-Frame-Options "SAMEORIGIN";
add_header X-XSS-Protection "1; mode=block";
add_header X-Content-Type-Options "nosniff";

# Advanced Proxy Config
send_timeout 5m;
proxy_read_timeout 120;
proxy_send_timeout 120;
proxy_connect_timeout 120;

# Basic Proxy Config
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Forwarded-Protocol $scheme;
proxy_set_header X-Forwarded-Host $http_host;
proxy_set_header X-Forwarded-Uri $request_uri;
proxy_set_header X-Forwarded-Ssl on;
proxy_redirect  http://  $scheme://;
proxy_http_version 1.1;
proxy_cache_bypass $cookie_session;
proxy_no_cache $cookie_session;
proxy_buffers 64 256k;

set_real_ip_from 10.7.0.0/16;
real_ip_header X-Forwarded-For;
real_ip_recursive on;
```

Trimmed default.conf for nginx:
```

map $http_upgrade $connection_upgrade {
  default upgrade;
  '' close;
}

server {
  listen 80 default_server;
  listen [::]:80 default_server;
  server_name _;

  proxy_headers_hash_max_size 512;
  proxy_headers_hash_bucket_size 128; 

  location ~/motioneye/?(.*)?$ {
    # access_log off;
    add_header RP2 true always;
    if ($request_uri ~ /motioneye$ ) {
      add_header Content-Type text/html;
      return 200 '<meta http-equiv="refresh" content="0; URL=$uri/" />';
    }
    # add_header Content-Type text/html;
    # return 200 'hi';

    # Auto-login
    add_header Set-Cookie meye_password_hash=XXXXXXXXXXXXXXXXXXXXXX;
    add_header Set-Cookie meye_username=XXXXXXXXX;

    if ($query_string ~ noauth=) {
      add_header Set-Cookie meye_password_hash="";
      add_header Set-Cookie meye_username="";
    }

    if ($query_string ~ nologin=) {
      add_header Set-Cookie meye_password_hash="";
      add_header Set-Cookie meye_username="";
    }

    if ($query_string ~ logout=) {
      add_header Set-Cookie meye_password_hash="";
      add_header Set-Cookie meye_username="";
    }

    proxy_pass http://motioneye.home.svc.cluster.local;
    rewrite /motioneye/(.*) /$1 break;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    include conf.d/proxy.config;
    include conf.d/auth.config;
  }
```

There's an nginx ingress controller handing https and certs, and a nginx reverse proxy that passes incoming requests to this nginx reverse proxy that handles Authelia auth. But this should not interfere with testing with the curl command.

I use Metal-LB for networking.

**Comment by james-d-elliott** [comment]
So my initial thoughts are this is a networking issue. What is 10.7.9.10 and why is it getting an i/o timeout on both NTP and LDAP? It's a commonly used rule in technology to start on the lowest possible layer of the OSI model first. This is also indicated by the 502 Bad Gateway response you're getting from the auth_request.

Nothing here is conclusive enough to say this is a bug with Authelia, there's a bad configuration somewhere and I'd guess it's routing related.

**Comment by Slyke** [comment]
I suspect you're right. I've noticed it depends which node the spawns on when certain issues. For example, when my worker2 has the reverse proxy on it, it can't communicate with motioneye and all grafana routes return 404. If it spawns on worker1 everything works. It likes to spawn on worker2 most of the time though. No idea why this has started happening, but either way, it's not Authelia's fault. Sorry for the trouble!

**Comment by james-d-elliott** [comment]
No worries, let us know if we can assist either way, just because it doesn't seem on the surface to be Authelia doesn't mean it can't be. Plus we may be able to help.

After thinking about it I'm wondering if your LAN subnets are conflicting with your cluster subnets in some way.

**Comment by Slyke** [comment]
Thanks haha. I think it's actually the Ingress controller. I've been chasing this issue down for almost a week now. I've discovered that the IP addresses of services are updating every second or so (being replaced with the ingress controller IP, or their Metal-LB assigned IP) which is why it was super hard to figure out what was going on. It would get router back through the ingress controller, with all the headers to be passed back through nginx to my service, which had a lets say 50% chance to either pass it back to the ingress controller or the actual service depending on the IP at the time, each time.

I posted about it here if you wanted to keep tabs: https://github.com/kubernetes/ingress-nginx/issues/8539 Still investigating it myself.
